### ISSUE TITLE
TERRA-52 ⁃ astra_role failure json: cannot unmarshal string into Go value of type astra.Errors

### ISSUE THREAD

**Issue by ndemeshchenko** [datastax/terraform-provider-astra#77]
Hi there,

here is the error while creating role

```
astra_role.role: Creating...
╷
│ Error: json: cannot unmarshal string into Go value of type astra.Errors
│
│   with astra_role.role,
│   on astra-db.tf line 49, in resource "astra_role" "role":
│   49: resource "astra_role" "role" {
│
```

### Terraform Version
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/datastax/astra v2.0.1

### Affected Resource(s)
- astra_role

### Terraform Configuration Files
```hcl
resource "astra_role" "role" {
  role_name   = "name"
  description = "desc"
  effect      = "allow"
  resources = [
    "drn:astra:org:${var.ds_org_id}"
  ]
  policy = ["accesslist-read", "db-all-keyspace-describe", "db-keyspace-describe", "db-table-select", "db-table-describe", "db-graphql", "db-rest", "db-cql"]
}
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by emerkle826** [comment]
@ndemeshchenko Can you try upgrading to version 2.0.3 of the terraform plugin? It has some improvements around this unmarshal error. It probably won't fix your issue, but it may give a little better indication of what is wrong.

**Comment by kenphused** [comment]
Got feedback from a DS customer who found that the problem was due to insufficient permissions on the token he was using.  He suggests better error handing / logging wrt token permissions.

**Comment by emerkle826** [comment]
Closing this as it was resolved in v2.0.3
